Upgrade to Expo 44

brettdh opened the pull request:
Includes iOS 15 support.

wittjosiah commented on src/navigation/MenuButton.js:
```diff
@@ -0,0 +1,17 @@
+import { Text } from 'react-native';
+
+const MenuButton = ({ navigation }) => (
+  <Text onPress={() => navigation.navigate('DrawerOpen')}>
```
Is `'DrawerOpen'` a constant that react-navigation is expecting?

brettdh replied:
That's how I understand it, but I'm just going from one or two examples myself.

wittjosiah commented on src/styles/index.js:
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
```
I'm not sure its a good idea to separate styles from the components. These are fairly basic, foundation styles so its probably okay, but in general I think its better to keep the styles with the component they are styling. This allows you to understand what a component will look like without having to jump back and forth between the styles and also prevents you from reusing the styles without thinking more carefully about the overlap between the components.

brettdh commented:
> I'm not sure its a good idea to separate styles from the components.

I agree in general. I just don’t yet have any examples of styles that are
tightly coupled to their components. I suspect those will come up soon,
though, when we start picking up the tasks in #12.

Also, we probably want a generic `Screen` component that contains the
styles common to all screens - and that’s probably where these `container`
styles actually belong.
On Sun, Jan 14, 2018 at 9:28 AM Josiah Witt <notifications@github.com>
wrote:

> *@wittjosiah* approved this pull request.
> ------------------------------
>
> In src/navigation/MenuButton.js
> <https://github.com/theliturgists/app/pull/11#discussion_r161399066>:
>
> > @@ -0,0 +1,17 @@
> +import React from 'react';
> +import PropTypes from 'prop-types';
> +import { Text } from 'react-native';
> +
> +const MenuButton = ({ navigation }) => (
> +  <Text onPress={() => navigation.navigate('DrawerOpen')}>
>
> Is 'DrawerOpen' a constant that react-navigation is expecting?
> ------------------------------
>
> In src/styles/index.js
> <https://github.com/theliturgists/app/pull/11#discussion_r161399362>:
>
> > @@ -0,0 +1,9 @@
> +import { StyleSheet } from 'react-native';
> +
> +export default StyleSheet.create({
>
> I'm not sure its a good idea to separate styles from the components. These
> are fairly basic, foundation styles so its probably okay, but in general I
> think its better to keep the styles with the component they are styling.
> This allows you to understand what a component will look like without
> having to jump back and forth between the styles and also prevents you from
> reusing the styles without thinking more carefully about the overlap
> between the components.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/theliturgists/app/pull/11#pullrequestreview-88684071>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFlnnJOlNDtcLrTseFObidyg0J0PuUeks5tKg7zgaJpZM4RWbOx>
> .
>

brettdh committed:
chore!: upgrade to expo 44

Includes iOS 15 support.

brettdh committed:
ci: update macos and xcode versions

brettdh committed:
build: upgrade detox and fix warning

brettdh committed:
build: downgrade flipper and remove workaround

brettdh committed:
ci: switch to actions/setup-java

brettdh committed:
build: fix detox builds by removing flipper

Couldn't get it to stop crashing the app on startup.
Removing it fixed the crashes.

brettdh committed:
ci: allow android detox tests to fail for now
